Design-Docs/General Design Doc.md

```diff
@@ -0,0 +1,131 @@
+* Drawer (Bottom, Orange)
+  * Year Slider (Far Left, Strawberry)
+  * Debt Type DD (Second from Left, Sea Green)
+  * Speed Toggle (Third from Left, Purple)
+  * Playback Button (Far Right, Pink)
+
+
+#### Component Relationship
+(Note: Some Relationships will not be displayed in the image, as concepts evolved over time)
+![Shit Tier Wireframe](Wireframes/Component_Relationship/US_State_Debt_Map_CR.png)
+* Drawer Toggle -> Drawer  (Below Map; Lavender)
+
+  The Drawer Toggle begans as a button style to be a down arrow, When the user clicks the down arrow, the drawer will pop down. Once the drawer is exposed, the Drawer Toggle button is then styled to be an up arrow. When the user clicks the up arrow, the drawer will dissapear, and the Up Arrow becomes a down arrow again. If the US Map is in the middle of auto-play, it will continue to play, but once the animation ends, auto-play will be disabled.
+  
+  * Secondary Actions
+  
+    Once the Drawer is closed, the US Map begins to auto-play every 10 seconds. 
+    
+* Year Slider -> Year; US Map (Drawer; Strawberry)
+
+  The Slider will directly affect the year and map animation simultaneously through manual control. The Slider will begin at 1999 and end at 2016. As the Slider is moved from left to right, the colors in US Map will change according to the year that is displayed.
+  * Secondary Actions
+  
+* Debt Type DD -> Header; US Map (Drawer; Sea Green)
+
+  The Debt Type DD contains a list of the different type of debts, once one is chosen, the header will change to reflect the selection. And the US Map will adjust to prepare for itself to playback that debt type.
+  
+* Speed DD -> US Map (Drawer; Purple)
+
+  The Speed DD contains a list of speeds, once one is chosen, the US Map will update its speed for the next playback.
+  
+* Playback -> US Map (Drawer; Pink)
+  
+  The Playback is a button that once pressed will trigger the US Map Animation based on the type and speed options that were selected.
+  
+  * Secondary Action
+    
+    The playback button is pressed, the slider should slide over accordingly.
+
+#### The Animation
+
+From 1999 to 2016 there is data on the different types of Debt. The site will show that data on the map using the colors which 
+will change as the years go up or down. The color transitions should relatively smooth from year to year. The year should be animated as well, such as a scroll up down.
+
+
+### Animated Table 
+![Shit Tier Wireframe](Wireframes/Photos/Animated_Table.jpg "Basic Wireframe")
+The Animated table is the second medium that will be used to show the user the SDD. The Table will have the States as the rows and the Years as the columns. There will be a drop down that has the different types of debt and the population. Depending on what is selected in the drop down, the table will display that data. 
+
+To help the user read the data easier, table will color code the data in different ways. 
+* Highlights By Year Using Timed Intervals
```

Comment:
The columns represent the years, so as the years progress, *all the states in that year* will be highlighted